avldb/src/main/scala/scorex/core/serialization/ManifestSerializer.scala

```diff
@@ -13,8 +12,8 @@ class ManifestSerializer(manifestDepth: Byte) extends ErgoSerializer[BatchAVLPro
   private val nodeSerializer = VersionedLDBAVLStorage.noStoreSerializer
 
   override def serialize(manifest: BatchAVLProverManifest[DigestType], w: Writer): Unit = {
-    val height = manifest.rootHeight
-    w.putBytes(Ints.toByteArray(height))
+    val rootNodeHeight = manifest.rootHeight.toByte
```

Comment:
No, by construction AVL+ tree has height <= 127 , then manifest has the same limitation also. Need to fix manifest in scrypto about that

Reply:
Yes, but it makes more assumption about input data. If however there is a mistake in the data, the silent overflow will lead to hard to localize bugs. Why not to do checks here? It will not impact performance.

Reply:
There MUST BE NOT mistakes here, if prover is having root height > 127 it can not work with the tree simply, forming bad manifest is very secondary issue then. Even more, height is much less than 127 always, there is a descriptive comment in scrypto about that.

Reply:
ok, I just curious why you think not checking overflow is better than checking?

Reply:
Because it could be more confusing actually for a reader to find a check for a thing which MUST NOT happen by construction. Conversion to int in some manifest / db classes is already confusing.